### ISSUE TITLE
`avoid_redundant_argument_values` false positive on `DateTime(year, month, day)` if month

### ISSUE THREAD

**Issue by noriHanda** [dart-lang/linter#3610]
**Describe the issue**

False positive occurs on `avoid_redundant_argument_values` when `month` and `day` are the same in `DateTime(year, month, day)`.

**To Reproduce**

```Dart
DateTime(2000, 1, 1)
```
the second `1` is marked by `avoid_redundant_argument_values`.

**Expected behavior**

`avoid_redundant_argument_values` does not check on `DateTime`.

**Comment by srawlins** [comment]
Oof, as much as I hate adding individual-function-specific exceptions (\*shudders*), I cannot in good conscience recommend to anyone that they write `DateTime(2000, 1)` or `DateTime(2000)` when trying to specify January 1, 2000. The [DateTime constructor](https://api.dart.dev/stable/2.17.7/dart-core/DateTime/DateTime.html) is pretty abnormal in this way. I'm not sure why the month and day are optional... it would save space if you need to create a bunch of DateTimes for January 1 of a several different years.

I'd be OK adding an exception to `avoid_redundant_argument_values` in the case of the function being `DateTime()`.

**Comment by bwilkerson** [comment]
I also hate adding exceptions to lint rules. The list of exceptions tends to grow until the lint rule is very hard to understand.

While I agree that `DateTime(2000, 1)` is hard to read, I'm wondering whether there might not be a more general solution (though I understand that `DateTime` is in `dart:core` and therefore adding an annotation to it is probably out of the question).

I might suggest, as a work-around, creating a utility method somewhere that requires all three values and returns a `DateTime` instance that's initialized.
